lib/aerospike/socket/ssl.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'openssl'
+
+module Aerospike
+  module Socket
+    class SSL < ::OpenSSL::SSL::SSLSocket
+      include Base
+
+      def self.connect(host, port, timeout, tls_name, ssl_options)
+        tcp_sock = TCP.connect(host, port, timeout)
+
+        ctx = OpenSSL::SSL::SSLContext.new
+        ctx.set_params(ssl_options) if ssl_options && !ssl_options.empty?
```

Comment:
This will fail if `ssl_options` includes any keys for which there are no corresponding assignment methods on `SSLContext`. E.g. if `ssl_options` includes a key `:enable`, this will raise an error because `set_params` will attempt to call `ctx.enable=`.

I think we should revert back to what you had before and support only a few dedicated keys in `ssl_options` that we use to call `ctx.add_certificate`. For anything more complicated than that we allow the user to pass in a fully configured SSLContext instance.

Reply:
Another option could be to only let `ssl_options` only represent options that can be passed to initialize the `SSLContext` skip the `enable` flag all together. If you temporarily want to disable SSL, you can just pass `nil` instead of the options. So:
- if `ssl_options` is a Hash, we create a new context and call `set_options`
- if `ssl_options` already is an `SSLContext` we just use that.

In a sense I think this is cleaner rather than maintaining a set of own options, because we can just refer to the official documentation for SSLContext. What do you think?